docs/android/emulator/troubleshooting.md

```diff
@@ -0,0 +1,475 @@
+---
+title: "Android Emulator Troubleshooting"
+description: "This article explains how to diagnose and work around problems that may occur when using the Android Emulator."
+zone_pivot_groups: platform
+ms.date: 12/03/2021
+---
+
+# Android emulator troubleshooting
+
+This article describes the most common warning messages and issues that occur while configuring and running the Android Emulator. Also, it describes solutions for resolving these errors and various troubleshooting tips to help you diagnose emulator problems.
+
+<!--
+::: zone pivot="windows"
+-->
+
+## Deployment issues on Windows
+
+Some error messages may be displayed by the emulator when you deploy your app. The most common errors and solutions are explained here.
+
+### Deployment errors
+
+If you see an error about a failure to install the APK on the emulator or a failure to run the Android Debug Bridge (**adb**), verify that the Android SDK can connect to your emulator. To verify emulator connectivity, use the following steps:
+
+01. Launch the emulator from the **Android Device Manager** (select your virtual device and select **Start**).
+
+01. Open a command prompt and go to the folder where **adb** is installed. If the Android SDK is installed at its default location, **adb** is located at _C:\\Program Files (x86)\\Android\\android-sdk\\platform-tools\\adb.exe_; if not, modify this path for the location of the Android SDK on your computer.
+
+01. Type the following command:
+
+    ```shell
+    adb devices
+    ```
+
+01. If the emulator is accessible from the Android SDK, the emulator should appear in the list of attached devices. For example:
+
+    ```shell
+    List of devices attached
+    emulator-5554   device
+    ```
+
+01. If the emulator doesn't appear in this list, start the **Android SDK Manager**, apply all updates, then try launching the emulator again.
+
+### MMIO access error
+
+If the message **An MMIO access error has occurred** is displayed, restart the emulator.
+
+<a name="gps-win"></a>
+
+## Missing Google Play Services
+
+If the emulated Android device doesn't have Google Play Services or Google Play Store installed, you probably created a virtual device that excluded these packages. When you [create a virtual device](device-manager.md), be sure to select one or both of the following options:
+
+- **Google APIs** &ndash; includes Google Play Services in the virtual device.
```

Comment:
Any "&ndash;" shows that the original content dates back to the days when the Xamarin publishing system couldn't handle the dash character!

Reply:
ndash is different from minus dash.

```
-  < dash
–  < en dash
—  < em dash
```

Don't know if it matters though. ..... OK I just looked it up in the style guide and I think in this case it should be an em dash `&mdash;` on these types of of thing—extra info.

I never really use them though, I can dump these if you want, or change them to em dash. I think it does look better though because it is a longer dash.

Reply:
My point, which I never explained(!) is that we don't need to specify them using escape codes. Anytime I encounter any dash escape codes I replace them with the an actual em dash character.